### PR TITLE
Create AudioDynamicTrigger

### PR DESCRIPTION
**Pull request by OrsonTyphanel93**
# Description : Dynamic Stacking of triggers : 

last Update ( 26, may 2024) : https://github.com/OrsonTyphanel93/adversarial-robustness-toolbox/blob/dev_1.14.0/The_END_Last_update_ART_dynamic_backdoor_attacks_trigger_stacking_(_Audio).ipynb

https://github.com/OrsonTyphanel93/adversarial-robustness-toolbox/blob/dev_1.14.0/Update_ART_dynamic_backdoor_attacks_trigger_stacking_(_Audio).ipynb

 "DynamicTrigger", exploits a "trigger stacking" technique combining numerous triggers to make detection more difficult. The model can learn to correlate the combined trigger with the desired output by stacking triggers. This means that even if the input has only one trigger, the model can anticipate the expected result. As a result, the model has the ability to generate identical samples with similar class titles, or to assign the label designated by the attacker for each sample, depending on its objectives. 

This code is designed to use the DynamicTrigger class to create a dynamic backdoor attack by generating a trigger and inserting it into an audio signal. If the generate_dynamic_trigger function correctly returns an instance of the DynamicTrigger class, then the attack works

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @OrsonTyphanel93 Thank you very much for your pull request! Could you please change the target branch to one of the dev branches?

**Comment by beat-buesser** [comment]
I'll change the target branch to `dev_1.18.0` which might induce merge conflicts.
